### PR TITLE
Fix #30936 - Toast should have fallback type

### PR DESCRIPTION
**Pull request by MariaAga**
for example Foreman_remote_execution uses flash[:notice] and causes page to render blank after importing a job template as it's not supported in PF
```
helpers.js:38 Uncaught Error: Unsupported alert type=notice
    at getClassName (helpers.js:38)
    at Alert (Alert.js:27)
    at renderWithHooks (react-dom.development.js:15108)
    at mountIndeterminateComponent (react-dom.development.js:17342)
    at beginWork$1 (react-dom.development.js:18486)
    at HTMLUnknownElement.callCallback (react-dom.development.js:347)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:397)
    at invokeGuardedCallback (react-dom.development.js:454)
    at beginWork$$1 (react-dom.development.js:23217)
    at performUnitOfWork (react-dom.development.js:22208)
14:15:39.103 react-dom.development.js:19814 The above error occurred in the <Alert> component:
    in Alert (created by ToastNotification)
    in ToastNotification (created by ToastsList)
    in div (created by ToastNotificationList)
    in ToastNotificationList (created by ToastsList)
    in ToastsList (created by ConnectFunction)
    in ConnectFunction (created by I18nProviderWrapper(Connect(ToastsList)))
    in IntlProvider (created by I18nProviderWrapper(Connect(ToastsList)))
    in I18nProviderWrapper(Connect(ToastsList)) (created by StoreProvider(I18nProviderWrapper(Connect(ToastsList))))
    in Provider (created by StoreProvider(I18nProviderWrapper(Connect(ToastsList))))
    in StoreProvider(I18nProviderWrapper(Connect(ToastsList)))
```
the ToastsList component should convert unknown types to info

### CODE REVIEW

File: webpack/assets/javascripts/react_app/components/ToastsList/ToastList.js [hunk 1]
```diff
@@ -16,11 +17,15 @@ const ToastsList = ({ messages, deleteToast }) => {
         ? ToastNotification
         : TimedToastNotification;
 
+      const type = Alert.ALERT_TYPES.includes(toastProps.type)
+        ? toastProps.type
+        : 'info';
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/ToastsList/ToastList.js]: I think it's a good idea to add a fallback, although maybe it's better to fix that flash method that passed a wrong type? I don't know what is the policy, should we also support default fallback to other components?
adding @sharvit @amirfefer @jeremylenz

Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/ToastsList/ToastList.js]: The reason for adding a fallback is that the whole page isn't shown when a wrong type is sent, I can add a deprecation warning or something but I would rather have a wrong color for the toast then no page at all

Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/ToastsList/ToastList.js]: Yeah I get it, though it can happen in other components too,
that's why I am trying to think what's the best solution

Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/ToastsList/ToastList.js]: I think it's a special case as it gets the props from a rails functionality, are other components like this?

Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/ToastsList/ToastList.js]: I think it's the same as mounting components from the server with `react_component`
so maybe in this case we should change `flash[:notice]` => `flash[:info]`
the supported types are `success/ warning/ error/ info` so like when using `props` I would assume the consumer will use one of them.

Comment (jeremylenz) [reply webpack/assets/javascripts/react_app/components/ToastsList/ToastList.js]: I would support adding the default alert type so the page can be displayed, but it should also `console.error` to bug the developer.  We shouldn't allow incorrect types to be called without impunity.

Comment (amirfefer) [reply webpack/assets/javascripts/react_app/components/ToastsList/ToastList.js]: A default alert type can create a weird UI,  for example -  someone accidentally has a typo  `flash[:errorr]` and the UI renders an `info` alert in return.  IMHO the page should be rendered with a default type but an error should be thrown as well, also a wrong flash type needs a fix regardless this PR - [these](https://github.com/search?q=org%3Atheforeman+flash%5B%3Anotice%5D&type=code) are the places which plugins use `flash[:notice]`

Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/ToastsList/ToastList.js]: I think we have an agreement here :+1:
also, after this fix maybe we should also contribute it to Patternfly (if not to v3 so maybe to v4)

I don't know what was the `notice` intention for, but I found this article: https://www.rubyguides.com/2019/11/rails-flash-messages/
mentioning: "I like to think about alert as an error message & a notice as a **confirmation message**."

Adding also @tbrisker @ekohl who might remember what it was :raised_hands:

Comment (tbrisker) [reply webpack/assets/javascripts/react_app/components/ToastsList/ToastList.js]: `notice` and `alert` are standard rails flash types, https://github.com/theforeman/foreman/blob/develop/app/controllers/concerns/foreman/controller/flash.rb#L5 adds `inline`, `success`, `info`, `warning` and `error` to the list (iirc this was done to match the patternfly types).
Looks like we deprecated and dropped `notice` in 1.20:

https://github.com/theforeman/foreman/commit/35c41f4d66ccb8094dd390b2c384e3dd80f4bc95#diff-773c68dec5799a1f613c256b208b90bdL7-L11 

But rails programmers who are used to using `notice` and wrote code since 1.20 will have no idea it was dropped since there is no deprecation message.
The plugins pointed out by @amirfefer are all inactive and archived, but `notice` flashes can be created using Rails methods as well: https://api.rubyonrails.org/classes/ActionDispatch/Flash/FlashHash.html#method-i-notice-3D

TL;DR: I think it makes sense to map `notice` to `info` just in case and maybe add a console warning about it in development. Failing when a type is not found is bad design on Patternfly's side, it should be fixed there but until it is we should provide a valid type to make sure pages don't fail to load, or use error boundaries so this error doesn't cause the full page failure (which might be a good idea in general to have an error boundary on the base component and display meaningful error messages when a page fails to load rather than blank screen)

Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/ToastsList/ToastList.js]: So to solve this issue we should:
* map notice to info and alert to warning and warn using console.warn
* everything else should be info and and warn using console.error
anything else?

**Comment by MariaAga** [comment]
Added console warn and error

**Review by tbrisker — COMMENTED** [review]
Please rebase on latest develop so katello tests pass

**Review by amirfefer — APPROVED** [review]
Thanks @MariaAga !
LGTM
